modules/unpickler/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/Scala3Unpickler.scala

```diff
@@ -498,8 +359,5 @@ class Scala3Unpickler(
             .getOrElse(regex.matches(javaType))
     rec(scalaType.toString, javaType.name)
 
-  private def skip(symbol: TermSymbol): Boolean =
-    (symbol.isGetterOrSetter && !symbol.isLazyValInTrait) || symbol.isSynthetic
-
-case class AmbiguousException(m: String) extends Exception
-case class NotFoundException(m: String) extends Exception
+  private def skip(bmthd: BinaryMethodSymbol): Boolean =
+    (bmthd.symbol.isDefined && bmthd.symbol.get.isGetterOrSetter && !bmthd.symbol.get.isLazyValInTrait) || bmthd.symbol.get.isSynthetic
```

Comment:
Pattern matching is probably more readable.

```suggestion
  private def skip(method: BinaryMethodSymbol): Boolean = method match
    case BinaryMethod(sym, Getter) => !sym.isLazyValInTrait
    case BinaryMethod(_, Setter) => true 
    case BinaryMethod(sym, _) if sym.isSynthetic => sym.isSynthetic
    case _ => false
```